### PR TITLE
Implemented configurable number of columns

### PR DESCRIPTION
**Pull request by marius7383**
I very much like how many features of flexbox gridlex embraces. In various projects I came across  the need to have different proportions / number of columns which is why I converted the hard-coded blocks into `@for` loops and introduced a variable called `$gl-colCount` which defaults to `12`. 
The file `gridlex-preprocessing.scss` uses this variable to calculate the column proportions and fractions which is then used in `gridlex-mixins.scss` to generate the needed column and grid definitions. 
I though I'd try to give this back for all the time I saved by not having to create a grid system by myself. Maybe you find it useful as well and would be willing to merge this PR.

In any case: Thanks for gridlex!

### CODE REVIEW

File: dist/gridlex.css [hunk 1]
```diff
@@ -509,116 +514,116 @@
...
+[class*="grid-"] > [data-push-left*="off-3"],
+[class*="grid_"] > [data-push-left*="off-3"] {
+  margin-left: 25%; }
+
 [class~="grid"] > [data-push-right*="off-3"],
 [class*="grid-"] > [data-push-right*="off-3"],
 [class*="grid_"] > [data-push-right*="off-3"] {
   margin-right: 25%; }
 
+[class~="grid"] > [data-push-left*="off-4"],
```
Comment (devlint) [inline dist/gridlex.css]: I have not look at all your work, but I see that offset are generated twice. Need to be fix before I decide to accept your great work :)

Comment (marius7383) [reply dist/gridlex.css]: I checked the diff of the CSS files again and it seems that it is not generated twice, but the order of rules between offset left and offset right is different if I put them in one for loop (as it is in this PR). If I seperate the loops as given below, the diff does not show any differences anymore. Would you like the loops to be seperate in order to have less differences?

    // Function to generate the offset
    @mixin makeOff($off)
    {
      [#{$gl-attributeName}~="#{$gl-gridName}"],
      [#{$gl-attributeName}*="#{$gl-gridName}-"],
      [#{$gl-attributeName}*="#{$gl-gridName}_"] {
        @for $i from 0 through $gl-colCount - 1 {
          > [data-push-left*="#{$off}-#{$i}"]{ margin-left: map-get($gl-colPortions, $i); }
        }
      }
    
      [#{$gl-attributeName}~="#{$gl-gridName}"],
      [#{$gl-attributeName}*="#{$gl-gridName}-"],
      [#{$gl-attributeName}*="#{$gl-gridName}_"] {
        @for $i from 0 through $gl-colCount - 1 {
          > [data-push-right*="#{$off}-#{$i}"]{ margin-right: map-get($gl-colPortions, $i); }
        }
      }
    }

**Comment by marius7383** [comment]
I just reworked the PR such that the diffs become slim and to be able to see that the offset rules are generated only once.

**Comment by devlint** [comment]
Merge!
Thank you for your work !!

**Pull request merged**